vignettes/custom_priors.Rmd

```diff
@@ -0,0 +1,156 @@
+---
+title: Setting up custom priors
+description: "How to pass custom priors to model modules"
+author: Epinowcast Team
+output: 
+  bookdown::html_vignette2:
+    fig_caption: yes
+    code_folding: show
+pkgdown:
+  as_is: true
+csl: https://raw.githubusercontent.com/citation-style-language/styles/master/apa-numeric-superscript-brackets.csl
+vignette: >
+  %\VignetteIndexEntry{Setting up custom priors}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = TRUE
+)
+```
+
+The various modules in `epinowcast()`, that is `enw_reference()`, `enw_expectation()`, `enw_report()`, `enw_missing()`, and `enw_obs()` have default priors that are set internally. However, these can be customised.
+
+This vignette explains how to define custom priors for the various modules in the `epinowcast()` function.
+
+# Default priors
+
+We will start by describing the default priors for each model component. This will help you understand the structure of the priors and how to customise them.
+
+## Reference date module
+
+The reference date model module has the following priors.
+```{r reference_prior, echo = FALSE}
+enw_reference_prior <- data.frame(
```

Comment:
is this pulled out of the module itself? For maintainability it seems like we would be better off showing how to find this out from the code directly vs duplicating?

Reply:
if we did it like this then we could leave looking at this as an exercise to the reader and thus focus on documenting them in their internal tables more clearly?

Reply:
> is this pulled out of the module itself? For maintainability it seems like we would be better off showing how to find this out from the code directly vs duplicating?

Yes, but although I didn't realise a better way would have been to do something like:

```r
 enw_reference(data = enw_example("preprocessed"))$priors
```

And replace `enw_reference` with the respective module.

We could just point this out in the doc for `prior` in `epinowcast()` and remove this vignette. This issue started with me wondering what the required variables were and couldn't find them documented anywhere.

> if we did it like this then we could leave looking at this as an exercise to the reader and thus focus on documenting them in their internal tables more clearly?

Yes, we can just show them now to print out the table as I've stated above.